### DISCUSSION TITLE
Proxying to proxy ... how?

### DISCUSSION THREAD

**Discussion by InputOutput** [chimurai/http-proxy-middleware#551] [Q&A]
### Expected behavior
I want to have requests between our middleware using Express & http-proxy-middleware to be proxied to another debugging proxy (in this case OWASP ZAP) to be able to easily debug the traffic between middleware and API.
### Actual behavior
I am unable to find the correct setup to achieve this. I have activated the toProxy, xvfb and changeOrigin options but have no idea how to actually tell the proxy to proxy via OWASP.
### Setup

* http-proxy-middleware: 0.17.4
* server: express 4.16.2
* other relevant modules

#### proxy middleware configuration
```javascript
const proxyDefaults = {
    target: `${config.api.root}`,
    toProxy: true,
    xfwb: true,
    logLevel: config.loglevel.proxy,
    changeOrigin: true,
    secure: config.api.secure,
    pathRewrite: {
        '^/express': '/v2'
    },
    logProvider: function() {
        return logger;
    }
};
```
#### server mounting
```javascript
router.all('*', authenticationMiddleware, proxy(Object.assign({}, injectAccessToken, proxyDefaults)));

app.use('/express', router);
```

**Comment by markbjerke** [comment]
I would settle for using Fiddler for monitoring the usage of this package.  I agree with you there doesn't seem to be enough config since the toProxy is a boolean?

**Comment by AlbinoDrought** [comment]
Here's what I used to forward to OWASP ZAP, using the `http-proxy-agent` package (also see `node-proxy-agent` for SOCKS, PAC, HTTPS proxy support). I didn't actually need the `toProxy` option, not sure why, but I successfully see the requests in ZAP anyways.

```js
const express = require('express');
const proxy = require('http-proxy-middleware');
const ProxyAgent = require('http-proxy-agent');

// proxy to chain through (OWASP ZAP)
const proxyUrl = 'http://127.0.0.1:8000';

const app = express();

app.use(proxy({
  agent: new ProxyAgent(proxyUrl),
  target: 'http://example.com/',
  changeOrigin: true,
}));
```

**Comment by aquillano** [comment]
@AlbinoDrought I tried your solution to proxy to `mitmproxy`, but it didn't work. I wonder what the difference is between ZAP ant `mitmproxy`?
